From 004edb6684db3900b2218c693af186f3c367e511 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Mon, 20 Jan 2020 15:52:02 +0100
Subject: [PATCH] iio: tsl2x7x: Fix trigger, set defaults

Interrupt trigger is low and not rising.
Set 0x11 as default for Persistence Register 0xC.
Set proximity pulse count to 2, Proximity Pulse Count register 0xE.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 drivers/staging/iio/light/tsl2x7x.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/iio/light/tsl2x7x.c b/drivers/staging/iio/light/tsl2x7x.c
index 786e93f16ce9..9d9d429659cf 100644
--- a/drivers/staging/iio/light/tsl2x7x.c
+++ b/drivers/staging/iio/light/tsl2x7x.c
@@ -240,12 +240,12 @@ static const struct tsl2x7x_settings tsl2x7x_default_settings = {
 	.als_cal_target = 150,
 	.als_thresh_low = 200,
 	.als_thresh_high = 256,
-	.persistence = 255,
+	.persistence = 0x11,
 	.interrupts_en = 0,
 	.prox_thres_low  = 0,
 	.prox_thres_high = 512,
 	.prox_max_samples_cal = 30,
-	.prox_pulse_count = 8
+	.prox_pulse_count = 1
 };
 
 static const s16 tsl2X7X_als_gainadj[] = {
@@ -1905,7 +1905,7 @@ static int tsl2x7x_probe(struct i2c_client *clientp,
 		ret = devm_request_threaded_irq(&clientp->dev, clientp->irq,
 						NULL,
 						&tsl2x7x_event_handler,
-						IRQF_TRIGGER_RISING |
+						IRQF_TRIGGER_LOW |
 						IRQF_ONESHOT,
 						"TSL2X7X_event",
 						indio_dev);
-- 
2.16.4

